kernel/ps2/src/lib.rs

```diff
@@ -1,484 +1,704 @@
+    #[allow(dead_code)]
+    should_be_zero: u1,
+    /// disables the keyboard
+    port1_clock_disabled: bool1,
+    /// disables the auxilary device (mouse)
+    port2_clock_disabled: bool1, //NOTE: only if 2 PS/2 ports supported
+    /// whether IBM scancode translation is enabled (0=AT, 1=PC)
+    #[allow(dead_code)]
+    port1_translation_enabled: bool1,
+    #[allow(dead_code)]
+    must_be_zero: u1,
 }
 
 /// read the config of the ps2 port
-pub fn ps2_read_config() -> u8 {
-    ps2_write_command(0x20);
-    let config = ps2_read_data();
-    config
+fn read_config() -> ControllerConfigurationByte {
+    write_command(ReadFromInternalRAMByte0);
+    ControllerConfigurationByte::from_bytes([read_data()])
 }
 
 /// write the new config to the ps2 command port (0x64)
-pub fn ps2_write_config(value: u8) {
-    ps2_write_command(0x60);
-    unsafe { PS2_PORT.lock().write(value) };
+fn write_config(value: ControllerConfigurationByte) {
+    write_command(WriteToInternalRAMByte0);
+    write_data(WritableData::Configuration(value));
 }
 
 /// initialize the first ps2 data port
 pub fn init_ps2_port1() {
-    //disable PS2 ports first
-    ps2_write_command(0xA7);
-    ps2_write_command(0xAD);
-
-    // clean the output buffer of ps2 to avoid conflicts
-    ps2_clean_buffer();
+    init_ps2_port(PS2Port::One);
+}
 
-    // set the configuration
-    {
-        // read the old configuration
-        let old_config = ps2_read_config();
+/// initialize the second ps2 data port
+pub fn init_ps2_port2() {
+    init_ps2_port(PS2Port::Two);
+}
+enum PS2Port {
+    One,
+    Two,
+}
 
-        // set the new configuration
-        let new_config = (old_config & 0xEF) | 0x01;
-        ps2_write_config(new_config);
+// see https://wiki.osdev.org/%228042%22_PS/2_Controller#Initialising_the_PS.2F2_Controller
+fn init_ps2_port(port: PS2Port) {
+    // Step 3: Disable Devices
+    write_command(DisablePort2);
+    write_command(DisablePort1);
+
+    // Step 4
+    flush_output_buffer();
+
+    // Step 5: Set the Controller Configuration Byte
+    let mut config = read_config();
+    match port {
+        PS2Port::One => {
+            config.set_port1_clock_disabled(false);
+            config.set_port1_interrupt_enabled(true);
+        }
+        PS2Port::Two => {
+            config.set_port2_clock_disabled(false);
+            config.set_port2_interrupt_enabled(true);
+        }
     }
+    write_config(config);
+
+    // Step 9: Enable Devices
+    write_command(EnablePort1);
+    match port {
+        PS2Port::Two => write_command(EnablePort2),
+        PS2Port::One => (),
+    };
+}
 
-    ps2_write_command(0xAE);
+/// Clean the [PS2_DATA_PORT] output buffer, skipping the [ControllerToHostStatus] `output_buffer_full` check
+fn flush_output_buffer() {
+    //NOTE(hecatia): on my end, this is always 250 for port 1 and 65 for port 2, even if read multiple times
+    debug!("ps2::flush_output_buffer: {}", read_data());
```

Comment:
is this note something we should be aware of? I'm not sure what to think when I read it, is it a to-do item or a form of unexpected behavior?

Also probably should remove this debug statement since it's quite minor and seems to be unimportant (?)